crates/astria-sequencer/scripts/testnet.sh

```diff
@@ -16,27 +25,38 @@ for i in $(seq 0 "$((NUM_VALIDATORS - 1))"); do
     COMETBFT_P2P_PORT=$((28660 + $i))
 
     # update sequencer app port
-    sed -i "s/proxy_app = \"tcp:\/\/127.0.0.1:26658\"/proxy_app = \"tcp:\/\/127.0.0.1:$APP_PORT\"/g" $OUT_DIR/node$i/config/config.toml
+    edit_in_place "s/proxy_app = \"tcp:\/\/127.0.0.1:26658\"/proxy_app = \"tcp:\/\/127.0.0.1:$APP_PORT\"/g" $OUT_DIR/node$i/config/config.toml
     # RPC laddr
-    sed -i "s/laddr = \"tcp:\/\/127.0.0.1:26657\"/laddr = \"tcp:\/\/127.0.0.1:$COMETBFT_RPC_PORT\"/g" $OUT_DIR/node$i/config/config.toml
+    edit_in_place "s/laddr = \"tcp:\/\/127.0.0.1:26657\"/laddr = \"tcp:\/\/127.0.0.1:$COMETBFT_RPC_PORT\"/g" $OUT_DIR/node$i/config/config.toml
     # p2p laddr
-    sed -i "s/laddr = \"tcp:\/\/0.0.0.0:26656\"/laddr = \"tcp:\/\/127.0.0.1:$COMETBFT_P2P_PORT\"/g" $OUT_DIR/node$i/config/config.toml
+    edit_in_place "s/laddr = \"tcp:\/\/0.0.0.0:26656\"/laddr = \"tcp:\/\/127.0.0.1:$COMETBFT_P2P_PORT\"/g" $OUT_DIR/node$i/config/config.toml
 
     # update p2p persistent peer ports
     for j in $(seq 0 "$((NUM_VALIDATORS - 1))"); do
         P2P_PORT=$((28660 + $j))
-        sed -i "s/node$j:26656/127.0.0.1:$P2P_PORT/g" $OUT_DIR/node$i/config/config.toml
+        edit_in_place "s/node$j:26656/127.0.0.1:$P2P_PORT/g" $OUT_DIR/node$i/config/config.toml
     done
 
     # allow peers from the same IP
-    sed -i 's/allow_duplicate_ip = false/allow_duplicate_ip = true/' $OUT_DIR/node$i/config/config.toml
-    sed -i 's/addr_book_strict = true/addr_book_strict = false/' $OUT_DIR/node$i/config/config.toml
+    edit_in_place 's/allow_duplicate_ip = false/allow_duplicate_ip = true/' $OUT_DIR/node$i/config/config.toml
+    edit_in_place 's/addr_book_strict = true/addr_book_strict = false/' $OUT_DIR/node$i/config/config.toml
+
+    # update genesis file with app state
+    ../../target/debug/astria-sequencer-utils --genesis-app-state-file=test-genesis-app-state.json --destination-genesis-file=$OUT_DIR/node$i/config//genesis.json --chain-id=astria
+    sed -i'.bak' 's/timeout_commit = "1s"/timeout_commit = "2s"/g' ~/.cometbft/config/config.toml
 done
 
 # start sequencer application for each node
 for i in $(seq 0 "$((NUM_VALIDATORS - 1))"); do
-    APP_PORT=$((26660 + $i))
-    ../../target/debug/astria-sequencer --genesis-file=test-genesis.json --listen-addr=127.0.0.1:$APP_PORT &> $OUT_DIR/node$i/sequencer.log &
+    # edit default .env files for multiple validators 
+    DB_PATH="/tmp/astria_db_$i"
+    ASTRIA_SEQUENCER_LISTEN_ADDR=127.0.0.1:$((26660 + $i))
+    ASTRIA_SEQUENCER_GRPC_ADDR=127.0.0.1:$((8080 + $i))
+
+    export ASTRIA_SEQUENCER_DB_FILEPATH=$DB_PATH \
```

Comment:
We have the `local.example.env` file for this,`just copy-env` and such to configure them. I'd prefer not to have more ways that env vars are set which can create confusion.

Reply:
we'd have to manually make a new .env file for each node in that case

Reply:
I'd have to have the script edit those variables even with copying. This file is pretty brittle in general

Reply:
yeah just leave as-is, this is fine